File: update-simple/src/main/java/com/mqd/updatesimple/core/UpdateLibContext.java
package com.mqd.updatesimple.core;

import android.content.Context;

/**
 * 全局 Context 持有器。
 */
public class UpdateLibContext {
    private static Context context;
    private static boolean initialized = false;

    public static void init(Context ctx) {
        if (!initialized) {
            context = ctx.getApplicationContext();
            initialized = true;
        }
    }

    public static Context getContext() {
        return context;
    }

    public static boolean isInitialized() {
        return initialized;
    }
}

File: update-lib/src/main/java/com/mqd/updatelib/core/UpdateLibContext.kt
package com.mqd.updatelib.core

import android.annotation.SuppressLint
import android.content.Context

/**
 * 全局 Context 持有器
 *
 * @author quzhuligpt@gmail.com
 * @since 2026/6/29
 */
object UpdateLibContext {
    @SuppressLint("StaticFieldLeak")
    private lateinit var context: Context

    val isInitialized: Boolean
        get() = ::context.isInitialized

    @JvmStatic
    fun init(ctx: Context) {
        if (!isInitialized) {
            context = ctx.applicationContext
        }
    }

    @JvmStatic
    fun getContext(): Context {
        check(isInitialized) {
            "UpdateLibContext is not initialized. Call UpdateLibContext.init(context) first."
        }
        return context
    }
}
